python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -683,6 +683,16 @@ def applyAltMaskPlanes(self, mask, altMaskSpans):
                              and list of SpanSets to apply to the mask
         """
         for plane, spanSetList in altMaskSpans.items():
+
+            if (plane == "NO_DATA" and self.config.doUsePsfMatchedPolygons and
+                    "NO_DATA" in self.config.badMaskPlanes):
```

Comment:
I think you should move this block outside the `for` loop, as I'm worried about doing something that would only be undone on a subsequent iteration (or vice-versa).